weather_dl/download_pipeline/clients.py

```diff
@@ -233,6 +233,19 @@ def download(self, res: t.Dict, target: str) -> None:
         self.c.download(res, target)
 
 
+class PublicECMWFServerExtended(MARSECMWFServiceExtended):
```

Comment:
This does not work because `api.ECMWFService` requires `service` as an init arg which is not applicable in the case of the public datasets. I changed it to this to make it work:

```python
class PublicECMWFServerExtended(api.ECMWFDataServer):
    def __init__(self, *args, dataset='', **kwargs):
        super().__init__(*args, **kwargs)
        self.c = SplitMARSRequest(
            self.url,
            "datasets/%s" % (dataset,),
            email=self.email,
            key=self.key,
            log=self.log,
            verbose=self.verbose,
        )
        
    def fetch(self, req: t.Dict) -> t.Dict:
        return self.c.fetch(req)

    def download(self, res: t.Dict, target: str) -> None:
        self.c.download(res, target)
```

Reply:
Noted. Just made a change using a similar approach. PTAL.